x-pack/plugins/alerting/server/plugin.ts

```diff
@@ -403,7 +404,9 @@ export class AlertingPlugin {
           : Promise.resolve([]);
     });
 
-    scheduleAlertingTelemetry(this.telemetryLogger, plugins.taskManager);
+    this.eventLogService!.isEsContextReady().then(() => {
```

Comment:
> This promise returns a boolean indicating if startup was successful - should we use that and do something with it?

Maybe add the check and skip all the telemetry related to the Event Log index if the value is FALSE? What do you think?

> I'm not sure if it should gate scheduling the telemetry task?

Maybe I'm wrong, but I think we should wait with the alerting/actions telemetry task schedule till event log index will be initialized. Current bug happened because there were no shards to run the query on.
@pmuellr do you have any other thoughts how to pre event this failure?

> I'm also not quite sure I understand why the event log plugin has anything to do with this check?

That's because we are creating Event Log index template and all the stuff around this during Event Log plugin setups.
To query the new index we need make sure it it ready

Reply:
Oh wait, is the cause of the error from searching the event log index itself? Or does the error happen for any other index?

Reply:
The error happens only for event log index, because it is not yet initialized. The rest of telemetry we are requesting from the `.kibana` index.

Reply:
Ah okay, then this makes sense now, thanks for that.

> Maybe add the check and skip all the telemetry related to the Event Log index if the value is FALSE? What do you think?

Yea I think we need to do something, otherwise more errors might appear later right?

Reply:
We have the "false" end of event log initialization handled for writing event log documents, by having them log to the Kibana log instead of the event log index.  But it doesn't look like we have an equivalent sort of check - or even wait - for the query APIs.

The basic idea of the boolean promise on the `isEsContextReady()` call (which is just a delegation to `esContext::waitTillReady()`) is that you're waiting for the initialization to complete, and it will indicate if the initialization was successful or not.  The failure state isn't very specific - it could be that there was some creating new indices, but older event log indices would still be available for searching.  We don't provide that much detail.  So I'm not sure that we should really "do anything different" when it returns false.  

This is making me think that perhaps we should just put one of the wait calls at the top of our query methods - we only have one now, presumably more in the future:

https://github.com/elastic/kibana/blob/903e75ee03203337441fd492e568bb999eb47952/x-pack/plugins/event_log/server/es/cluster_client_adapter.ts#L341-L346

If we did that, would that resolve the telemetry issue?  Generally I'd prefer that folks not have to do too much poking around the event log in their own code - like waiting for initialization.  If at all possible we should put details like that into the event log itself.

I think we can just add a `this.wait()` to the top of the method - it seems like that's the same promise waiting for all the initialization to proceed.

Reply:
Anyway, we have a dependency on the Event log initialization before we start the telemetry task for alerting and actions. I think delaying it's registration will be better than increasing the task timeout.

@pmuellr do you think we should provide specific event log API for handling such a kind of aggregations (some generic API method)?

Reply:
Ah - we aren't using that `query*()` calls - I see.

As you know, since you already wrote an agg API for eventLog (but never merged it), yes we want one :-).  I thought we'd get one from RAC, but I guess not yet.  

If you want to add a simple one that works for your case, that seems fine.  Especially if it would clean up some of the explicit waiting.  We could later expand on that agg API for other cases.

Reply:
Did this conversation lead to a change in the code in this PR? I'm okay with how it is for now - I just didn't understand initially

Reply:
I'm honestly not sure if it's worth adding an agg API on event log just to avoid having an explicit "wait" API available.  Would require a lot more tests to add the agg rather than the wait API.  I'm happy with an explicit "wait" for folks that need to do ES calls outside of what EL APi supports, for now.

Reply:
I have this changes. No worries. Will push this soon. :-)

Reply:
> I have this changes. No worries. Will push this soon. :-)

@pmuellr, @chrisronline based on the offline discussion we decided that exposing a new Event Log API will be overcomplicated for such a case by a couple of reasons:
1.  It is difficult to design the reusable non-telemetry specific shape of the aggregation API
2. The other problem is how to access that API: using EventLogClient won't be easy to integrate with the telemetry task and EventLogService doesn't seems to be a good fit as well for the fetch event information.

Do you think we can live with the current solution till the EventLog expose another way to do this queries?